scripts/quests/otherAreas/Test_My_Mettle.lua

```diff
@@ -0,0 +1,152 @@
+                    if quest:getVar(player, 'Wait') >= os.time() then
+                        return quest:event(124)
+                    end
+
+                    -- Start quest
+                    return quest:event(120, xi.items.POWER_SANDALS, xi.items.POWER_SANDALS)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [120] = function(player, csid, option, npc)
+                    -- Option returns the sum of the gil option selected
+                    -- and the hour option selected, anything other than
+                    -- these combinations is invalid
+                    local possibleOptions =
+                    {
+                        124, 136, 148, 160, 172,
+                        274, 286, 298, 310, 322,
+                        524, 536, 548, 560, 572,
+                        1024, 1036, 1048, 1060, 1072,
+                    }
+                    local found = false
+                    for _, possibleOption in ipairs(possibleOptions) do
+                        if option == possibleOption then
+                            found = true
+                            break
+                        end
+                    end
+                    -- Rejected the quest at some point in the dialogue
+                    if found == false then
+                        return
+                    end
+
+                    -- Determine gil and hour options selected
+                    local gilOptions = { [1] = 100, [2] = 250, [5] = 500, [10] = 1000 }
+                    local gil = gilOptions[math.floor(option / 100)]
+                    local hours = option - gil
+
+                    -- Calculate potential payout
+                    local payoutRatios = { [24] = 3, [36] = 2.5, [48] = 2, [60] = 1.5, [72] = 1.2 }
+                    local gilPayout = math.floor(payoutRatios[hours] * gil)
+
+                    quest:setVar(player, 'GilPayout', gilPayout)
+                    quest:setVar(player, 'HoursSelected', 0)
+                    quest:setVar(player, 'HourStarted', VanadielHour())
+                    quest:setVar(player, 'DayStarted', VanadielDayOfTheYear())
+                    quest:setVar(player, 'Prog', 1)
+                    quest:begin(player)
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return quest:getVar(player, 'Prog') == 1
+        end,
+
+        [xi.zone.SELBINA] =
+        {
+            ['Devean'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if not npcUtil.tradeHasExactly(trade, xi.items.POWER_SANDALS) then
+                        return
+                    end
+
+                    local hoursRemaining = calculateHoursRemaining(player)
+                    if hoursRemaining > 0 then
+                        -- Quest succeeded
+                        return quest:event(122)
```

Comment:
When the quest is an important step like finishing or starting the quest, you should use the function `progressEvent` instead of `event` to force that specific event to happen.

Reply:
Done, thanks!